### PR TITLE
120 append fs title@main

### LINKED ISSUE #120: Add title as a part of append_fs, append_encodings and append_src
### Feature description

In every module, we have to add `card$append_text("Filter State", "header3")` before the `append_fs` call.
Please make the title call as a part of the `appenf_fs` call and take into account when there is no filter state to display.

EDIT: apply this to `append_fs`, `append_encodings` and `append_src` also.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct.

### Contribution Guidelines

- [X] I agree to follow this project's Contribution Guidelines.

### Security Policy

- [X] I agree to follow this project's Security Policy.

### PR DESCRIPTION
**Pull request by mhallal1**
closes #120 
Automatically add title when `append_fs`, `append_encodings` and `append_src` are used.
When `append_fs` is empty, nothing is shown in the previewer.

### CODE REVIEW

File: R/TealReportCard.R [hunk 1]
```diff
@@ -40,14 +41,18 @@ TealReportCard <- R6::R6Class( # nolint: object_name_linter.
       attr_fs <- attr(fs, "formatted")
       checkmate::assert_character(attr_fs, null.ok = TRUE)
 
-      if (is.null(attr_fs)) {
+      if (length(fs) != 0) {
+        self$append_text("Filter State", "header3")
+      }
+
+      if (!is.null(attr_fs) & length(fs) != 0) {
```
Comment (gogonzo) [inline R/TealReportCard.R:48]:
```suggestion
      if (length(fs) != 0) {
```

File: R/TealReportCard.R [hunk 2]
```diff
@@ -40,14 +41,18 @@ TealReportCard <- R6::R6Class( # nolint: object_name_linter.
       attr_fs <- attr(fs, "formatted")
       checkmate::assert_character(attr_fs, null.ok = TRUE)
 
-      if (is.null(attr_fs)) {
+      if (length(fs) != 0) {
```
Comment (nikolas-burkoff) [inline R/TealReportCard.R:44]: worth adding about if length 0 then nothing is added to the card?

File: R/TealReportCard.R (hunk 1 above)
Comment (nikolas-burkoff) [inline R/TealReportCard.R:48]:
```suggestion
      if (!is.null(attr_fs) && length(fs) != 0) {
```

File: R/TealReportCard.R [hunk 3]
```diff
@@ -40,14 +41,18 @@ TealReportCard <- R6::R6Class( # nolint: object_name_linter.
...
+        self$append_text("Filter State", "header3")
+      }
+
+      if (!is.null(attr_fs) & length(fs) != 0) {
+        self$append_text(attr_fs, "verbatim")
+      } else if (is.null(attr_fs) & length(fs) != 0) {
```
Comment (nikolas-burkoff) [inline R/TealReportCard.R:50]:
```suggestion
      } else if (is.null(attr_fs) && length(fs) != 0) {
```

**Review by nikolas-burkoff — COMMENTED** [review]
A couple of minor comments from me, but looks good. I'll let @Polkas approve

File: R/TealReportCard.R (hunk 2 above)
Comment (mhallal1) [reply R/TealReportCard.R:45]: updated

File: R/TealReportCard.R [hunk 4]
```diff
@@ -40,14 +42,18 @@ TealReportCard <- R6::R6Class( # nolint: object_name_linter.
...
+      }
+
+      if (!is.null(attr_fs) && length(fs) != 0) {
+        self$append_text(attr_fs, "verbatim")
+      } else if (is.null(attr_fs) && length(fs) != 0) {
         self$append_text(yaml::as.yaml(fs, handlers = list(
           POSIXct = function(x) format(x, "%Y-%m-%d"),
           POSIXlt = function(x) format(x, "%Y-%m-%d"),
           Date = function(x) format(x, "%Y-%m-%d")
         )), "verbatim")
```
Comment (Polkas) [inline R/TealReportCard.R:56]:
```suggestion
      if (length(fs) != 0) {
        self$append_text("Filter State", "header3")
        if (!is.null(attr_fs)) {
          self$append_text(attr_fs, "verbatim")
        } else if (is.null(attr_fs)) {
          self$append_text(yaml::as.yaml(fs, handlers = list(
            POSIXct = function(x) format(x, "%Y-%m-%d"),
            POSIXlt = function(x) format(x, "%Y-%m-%d"),
            Date = function(x) format(x, "%Y-%m-%d")
          )), "verbatim")
```

**Review by Polkas — APPROVED** [review]
🚀

**Pull request merged**